### DISCUSSION TITLE
[Question] Are there simple examples?

### DISCUSSION THREAD

**Discussion by Erscroas** [Mapsui/Mapsui#2257] [Q&A] [answered]
Hi there,
I wanted to start working with MapsUI since it looks really amazing in the sample-project you provided.
But I had 1 question, are there any more simple examples, that work with MAUI and MVVM?

I tried a couple of things but they don't seem to work for me.
I know that I am doing things wrong, hence why asking the question.

Kind regards
Erscroas

**Comment by pauldendulk** [comment]
I will move to the discussion tab.

**Comment by pauldendulk** [comment]
@Erscroas There are the quickstarts and then the samples. Nothing other than that. There are many samples so, https://mapsui.com/samples/ I am sure there are quite a few that er fairly simple. There are categories of samples and underneath that spefic samples. Look at the SourceCode tab to see how you could use them. In a lot of cares you could paste the entire class in to your own code with a few modifications. 

You mention MVVM. The ideas is that you bind the MapControl.Map property to your ViewModel and working with the Map from the ViewModel should give you all the options you need..

**Comment by Erscroas** [comment]
Yeah I managed already to show the OpenStreetMap in the page but zooming and centering to a location doesn't seem to work for me. 
I read the FAQ page about showing a dot in on the West-Coast of Africa. And it did, but I couldn't fix it.
But where I am at now is that the map doesn't zoom and center on the location provided.
This is the code i have now
``` csharp
[ObservableProperty]
Map map;

[ObservableProperty]
MapControl mapControl;

public MainViewModel()
{
    MapControl = new MapControl();
    Map = new Map();

    MapControl.Map = Map;
}

[RelayCommand]
async Task Appearing()
{
    Map.Layers.Add(new TileLayer(KnownTileSources.Create(KnownTileSource.OpenStreetMap)));

    var location = await Geolocation.GetLocationAsync();

    if (location != null)
    {
        var point = new MPoint(location.Longitude, location.Latitude);
        var sphericalMercatorCoordinate = SphericalMercator.FromLonLat(point.X, point.Y).ToMPoint();

        Map.Home = n => n.CenterOnAndZoomTo(sphericalMercatorCoordinate, n.Resolutions[9]);
    }
}
```

**Reply by pauldendulk** [reply, marked as answer]
Map.Home is the default Viewport. If you want to navigate to a new viewport use Map.Navigator.CenterAndZoomTo. This is a very common problem, We should improve that design.

Look at the Projection samples to fix the West-Coast of Africa problem.

**Reply by Erscroas** [reply]
Thank you, when I changed the Map.Home to Map.Navigator it let everyhing work.
```csharp

[ObservableProperty]
Map map;

[ObservableProperty]
MapControl mapControl;

public MainViewModel()
{
    MapControl = new MapControl();
    Map = new Map();

    MapControl.Map = Map;
}

[RelayCommand]
async Task Appearing()
{
    Map.Layers.Add(new TileLayer(KnownTileSources.Create(KnownTileSource.OpenStreetMap)));

    var location = await Geolocation.GetLocationAsync();

    if (location != null)
    {
        var point = new MPoint(location.Longitude, location.Latitude);
        var sphericalMercatorCoordinate = SphericalMercator.FromLonLat(point.X, point.Y).ToMPoint();

        Map.Navigator.CenterOnAndZoomTo(sphericalMercatorCoordinate, Map.Navigator.Resolutions[16]);
    }
}

```
